File: apps/flutter/tool/visual_smoke.mjs
import { chromium } from '@playwright/test';
import { mkdir } from 'node:fs/promises';
import { resolve } from 'node:path';

const output = resolve('.design/flutter-app/screenshots');
await mkdir(output, { recursive: true });

const browser = await chromium.launch({ headless: true });
const cases = [
  { name: 'login-light-phone', width: 390, height: 844, scheme: 'light' },
  { name: 'login-dark-phone', width: 390, height: 844, scheme: 'dark' },
  { name: 'login-light-desktop', width: 1440, height: 960, scheme: 'light' },
  { name: 'login-dark-desktop', width: 1440, height: 960, scheme: 'dark' },
];

for (const visualCase of cases) {
  const page = await browser.newPage({
    viewport: { width: visualCase.width, height: visualCase.height },
    colorScheme: visualCase.scheme,
    reducedMotion: 'no-preference',
  });
  const errors = [];
  page.on('console', (message) => {
    if (
      message.type() === 'error' &&
      !message.text().includes('ERR_CONNECTION_REFUSED')
    ) {
      errors.push(message.text());
    }
  });
  await page.goto('http://127.0.0.1:7357', { waitUntil: 'networkidle' });
  await page.locator('flutter-view').waitFor({ timeout: 90_000 });
  await page.waitForTimeout(2_000);
  await page.screenshot({
    path: resolve(output, `${visualCase.name}.png`),
    fullPage: true,
  });
  if (errors.length > 0) {
    throw new Error(`${visualCase.name}: ${errors.join('\n')}`);
  }
  await page.close();
}

await browser.close();
console.log(`Captured ${cases.length} visual smoke screenshots in ${output}`);

File: apps/flutter/tool/visual_smoke.py
import asyncio
import os
from pathlib import Path

from playwright.async_api import async_playwright


# The web build exercises only the responsive login surface. Asael native
# authentication intentionally rejects non-Android/iOS platforms; signed-in
# phone surfaces are covered by Flutter widget tests instead.


BASE_URL = os.environ.get("ASAEL_VISUAL_URL", "http://127.0.0.1:7357")
OUTPUT = Path(".design/flutter-app/screenshots")


async def capture_login(browser, name, width, height, color_scheme):
    page = await browser.new_page(
        viewport={"width": width, "height": height},
        color_scheme=color_scheme,
        reduced_motion="no-preference",
    )
    errors = []
    page.on(
        "console",
        lambda message: errors.append(message.text)
        if message.type == "error" and "ERR_CONNECTION_REFUSED" not in message.text
        else None,
    )
    await page.goto(BASE_URL, wait_until="networkidle")
    await page.locator("flutter-view").wait_for(timeout=90_000)
    await page.wait_for_timeout(750)
    await page.screenshot(path=OUTPUT / f"{name}.png", full_page=True)
    assert not errors, f"{name}: {'; '.join(errors)}"
    await page.close()


async def main():
    OUTPUT.mkdir(parents=True, exist_ok=True)
    async with async_playwright() as playwright:
        browser = await playwright.chromium.launch(headless=True)
        for case in [
            ("login-light-phone", 390, 844, "light"),
            ("login-dark-phone", 390, 844, "dark"),
            ("login-light-desktop", 1440, 960, "light"),
            ("login-dark-desktop", 1440, 960, "dark"),
        ]:
            await capture_login(browser, *case)
        await browser.close()


if __name__ == "__main__":
    asyncio.run(main())
